src/backend/distributed/utils/colocation_utils.c

```diff
@@ -1261,7 +1261,7 @@ DeleteColocationGroupLocally(uint32 colocationId)
 		 * AccessShareLock on replica identity index.
 		 */
 		Relation replicaIndex =
-			index_open(RelationGetReplicaIndex(pgDistColocation),
+			index_open(RelationGetPrimaryKeyIndex(pgDistColocation),
```

Comment:
comment is stale.

Also, why not `GetRelationIdentityOrPK `?

Reply:
We use `GetRelationIdentityOrPK ` when we don't know whether the table has primary key index or replica identity index. In this case, we know that `pg_dist_colocation` and `pg_dist_node` have primary key indexes.